### PR TITLE
improve specs per rubocop overlords

### LINKED ISSUE #194: Refactor testing for reliability and updated patterns
The test suite is significantly out of date and introduces several code smells that should be given attention to refactor. One result of this is tests that fail when run individually.

For instance, the full suite passes:

```
moab-versioning on  main [!] via 💎 v3.1.1 on ☁️  (us-west-2) took 6m22s 
❯ bundle exec rspec
.
.
.
Finished in 3.8 seconds (files took 0.38009 seconds to load)
403 examples, 0 failures

Coverage report generated for RSpec to /Users/amcollie/github/sul-dlss/moab-versioning/coverage. 1654 / 1691 LOC (97.81%) covered.
```

However, running the `storage_object_spec` alone fails:

```
❯ bundle exec rspec spec/unit_tests/moab/storage_object_spec.rb
.............F...WARNING: Using `expect { }.not_to raise_error(SpecificErrorClass)` risks false positives, since literally any other error would cause the expectation to pass, including those raised by Ruby (e.g. `NoMethodError`, `NameError` and `ArgumentError`), meaning the code you are intending to test may not even get reached. Instead consider using `expect { }.not_to raise_error` or `expect { }.to raise_error(DifferentSpecificErrorClass)`. This message can be suppressed by setting: `RSpec::Expectations.configuration.on_potential_false_positives = :nothing`. Called from /Users/amcollie/github/sul-dlss/moab-versioning/spec/unit_tests/moab/storage_object_spec.rb:362:in `block (3 levels) in <top (required)>'.
.........

Failures:

  1) Moab::StorageObject #size
     Failure/Error:
       Find.find(object_pathname) do |path|
         size += FileTest.size(path) unless FileTest.directory?(path)
       end
     
     NameError:
       uninitialized constant Moab::StorageObject::Find
     
             Find.find(object_pathname) do |path|
             ^^^^
     # ./lib/moab/storage_object.rb:230:in `size'
     # ./spec/unit_tests/moab/storage_object_spec.rb:474:in `block (2 levels) in <top (required)>'

Finished in 0.32443 seconds (files took 0.26354 seconds to load)
26 examples, 1 failure

Failed examples:

rspec ./spec/unit_tests/moab/storage_object_spec.rb:470 # Moab::StorageObject #size

Coverage report generated for RSpec to /Users/amcollie/github/sul-dlss/moab-versioning/coverage. 1182 / 1691 LOC (69.9%) covered.
```

This ticket may need to eventually be an epic to capture individual test refactors and issues.

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made? 🤔

OCD.   (I addressed some antipatterns pointed out by rubocop.)

Closes #194

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (e.g. changes what is written to filesystem, or what is expected to be read from filesystem), run ***[integration test create_preassembly_image_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** on stage as it tests preservation, and/or test in stage environment, in addition to specs.⚡

### CODE REVIEW

**Review by justinlittman — COMMENTED** [review]
In multiple locations in the tests, `expect ... with` were split into `allow ... with` and `expect ... receive`. Splitting `allow` and `expect` is consistent with our practices. However, also to be consistent, shouldn't it be `expect .. have_received`?

Happy to discuss.

**Comment by ndushay** [comment]
> In multiple locations in the tests, `expect ... with` were split into `allow ... with` and `expect ... receive`. Splitting `allow` and `expect` is consistent with our practices. However, also to be consistent, shouldn't it be `expect .. have_received`?
> 
> Happy to discuss.

Definitely true and my objective;  changing from "received" to "have_received" hits a LOT of tests so after I got this far yesterday, I decided it was enough for a PR.  I thought I would work on `have_received` today.

**Comment by ndushay** [comment]
And now, it has been addressed in this PR;  many of the rubocop rspec violations have been addressed.

> > In multiple locations in the tests, `expect ... with` were split into `allow ... with` and `expect ... receive`. Splitting `allow` and `expect` is consistent with our practices. However, also to be consistent, shouldn't it be `expect .. have_received`?
> > Happy to discuss.
> 
> Definitely true and my objective; changing from "received" to "have_received" hits a LOT of tests so after I got this far yesterday, I decided it was enough for a PR. I thought I would work on `have_received` today.

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -78,59 +71,18 @@ Naming/VariableNumber:
...
-# Offense count: 133
-RSpec/MultipleExpectations:
-  Max: 12
-
-# Offense count: 6
+# Offense count: 3
 # This cop supports safe autocorrection (--autocorrect).
 RSpec/ScatteredLet:
   Exclude:
     - 'spec/unit_tests/moab/file_group_difference_spec.rb'
```
Comment (ndushay) [inline .rubocop_todo.yml:84]: What remains is easier to read as is.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**